extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -235,19 +255,23 @@ class ConnectorsManager {
                 this._metricsHandler.onConnectorDestroyed();
```

Comment:
should call `RetainBucketsDecorator.onConnectorUpdated(connector)` around here, so that the buckets marked as retained can be cleared